astropy/utils/masked/core.py

```diff
@@ -990,6 +990,11 @@ def max(self, axis=None, out=None, **kwargs):
             axis=axis, out=out, **self._reduce_defaults(kwargs, np.nanmin)
         )
 
+    def ptp(self, axis=None, out=None, **kwargs):
+        result = self.max(axis=axis, out=out, **kwargs)
+        result -= self.min(axis=axis, **kwargs)
+        return result
```

Comment:
Is there a need to store intermediate variables and not just go for the one line?
```suggestion
        return self.max(axis=axis, out=out, **kwargs) - self.min(axis=axis, **kwargs)
```

Reply:
I had that at first, but it is wrong: we need to propagate a possible `out`, so the subtraction has to be in-place.